core/native/src/main/scala/me/shadaj/scalapy/interpreter/Platform.scala

```diff
@@ -57,4 +60,12 @@ object Platform {
   def setPtrLong(ptr: Pointer, offset: Int, value: Long): Unit = !((ptr + offset)).asInstanceOf[Ptr[Long]] = value
   def setPtrInt(ptr: Pointer, offset: Int, value: Int): Unit = !((ptr + offset)).asInstanceOf[Ptr[Int]] = value
   def setPtrByte(ptr: Pointer, offset: Int, value: Byte): Unit = !((ptr + offset)).asInstanceOf[Ptr[Byte]] = value
+
+  def programName: Option[Ptr[CWideChar]] =
+    Properties.propOrNone("scalapy.python.programname")
+      .orElse(sys.env.get("SCALAPY_PYTHON_PROGRAMNAME"))
+      .map(toCWideString(_))
+
+  private def toCWideString(str: String): Ptr[CWideChar] =
+    CPythonAPI.Py_DecodeLocale(Zone(implicit zone => toCString(str)), null)
```

Comment:
The resulting pointer needs to be freed. Perhaps change this function to instead take a callback, so that you can decode, run the callback, and then free with `PyMem_RawFree()`?